clippy_lints/src/future_not_send.rs

```diff
@@ -59,7 +59,21 @@ declare_clippy_lint! {
     "public Futures must be Send"
 }
 
-declare_lint_pass!(FutureNotSend => [FUTURE_NOT_SEND]);
+impl_lint_pass!(FutureNotSend => [FUTURE_NOT_SEND]);
+
+pub(crate) struct FutureNotSend {
+    future_trait: Option<DefId>,
+    send_trait: Option<DefId>,
+}
+
+impl FutureNotSend {
+    pub(crate) fn new(tcx: TyCtxt<'_>) -> Self {
+        Self {
+            future_trait: tcx.lang_items().future_trait(),
```

Comment:
This seems overkill to me, `.future_trait()` is only a fixed offset table access, it doesn't use any hashing or things like that.